src/modules/gateway/sagas.ts

```diff
@@ -157,26 +166,75 @@ function* upsertPurchase(
 }
 
 function* handleStorageLoad() {
-  const pendingManaPurchase: ReturnType<typeof getPendingManaPurchase> = yield select(
-    getPendingManaPurchase
+  const pendingPurchases: ReturnType<typeof getPendingPurchases> = yield select(
+    getPendingPurchases
   )
 
-  if (pendingManaPurchase) {
-    const { network, gateway, id } = pendingManaPurchase
+  if (pendingPurchases) {
+    for (const pendingPurchase of pendingPurchases) {
+      const { network, gateway, id } = pendingPurchase
+      switch (gateway) {
+        case NetworkGatewayType.TRANSAK:
+          yield put(pollPurchaseStatusRequest(pendingPurchase))
+          break
+        case NetworkGatewayType.MOON_PAY:
+          yield put(
+            manaFiatGatewayPurchaseCompleted(
```

Comment:
Is this necessary? isn't it just updating the purchase to its same state?

Reply:
The default behavior of our MoonPay integration is to poll the order status until it reaches a final one. In this case, it will dispatch the `manaFiatGatewayPurchaseCompleted` action to use that same logic. It will start as a pending order and change to different statuses after fetching that information from MoonPay API.